src/app/treasure-hunt/treasure-hunt-calculator-services/water-heating-treasure-hunt.service.ts

```diff
@@ -93,6 +111,83 @@ export class WaterHeatingTreasureHuntService {
     return cardData;
 }
 
+getEnergyData(waterHeatingOpportunity: WaterHeatingTreasureHunt, settings: Settings, currentEnergyUsage: EnergyUsage, opportunitySummary: OpportunitySummary): {
+  annualEnergySavings: Array<{
+    savings: number,
+    label: string,
+    energyUnit: string
+  }>,
+  percentSavings: Array<{ percent: number, label: string, baselineCost: number, modificationCost: number }>,
+  utilityTypes: Array<string>
+}{
+  let annualEnergySavings: Array<{ savings: number, label: string, energyUnit: string }> = new Array();
+  let percentSavings: Array<{ percent: number, label: string, baselineCost: number, modificationCost: number }> = new Array();
+  let utilityTypes: Array<string> = new Array();
+  let output: WaterHeatingOutput = this.waterHeatingService.waterHeatingOutput.getValue();
+
+  if (waterHeatingOpportunity.energySourceData.energySourceType == 'Natural Gas') {
```

Comment:
I think the if conditions logic at 128/148 can be removed/reworked. I think we're able to set units and label straight from EnergySourceData.

The water portion is fine. When/if we get more Mixed opp calculators, modifying EnergySourceData to hold multiple energy types would be better.